sound/soc/sof/pm.c

```diff
@@ -335,8 +336,20 @@ int snd_sof_prepare(struct device *dev)
 		return 0;
 
 #if defined(CONFIG_ACPI)
-	if (acpi_target_system_state() == ACPI_STATE_S0)
+	system_target_state = acpi_target_system_state();
+	switch (system_target_state) {
+	case ACPI_STATE_S0:
 		sdev->system_suspend_target = SOF_SUSPEND_S0IX;
+		break;
+	case ACPI_STATE_S1:
+	case ACPI_STATE_S2:
+	case ACPI_STATE_S3:
+		sdev->system_suspend_target = SOF_SUSPEND_S3;
+		break;
+	default:
+		sdev->system_suspend_target = SOF_SUSPEND_S3;
+		break;
```

Comment:
Hrm, before this patch the `sdev->system_suspend_target` would be `SOF_SUSPEND_S3` unless the target state is ACPI_STATE_S0, in that case it will be SOF_SUSPEND_S0IX.

With this patch the `sdev->system_suspend_target` would be `SOF_SUSPEND_S3` unless the target state is ACPI_STATE_S0, in that case it will be SOF_SUSPEND_S0IX.

Or am I missing something?

Reply:
Oh, I see.

Reply:
I guess no change then?